caf: modules: ble_adv: Power down on disconnection with reason 0x15 (#13987)
nrfconnect/sdk-nrf: doc/nrf/libraries/caf/ble_adv.rst

   The :kconfig:option:`CONFIG_CAF_BLE_ADV_GRACE_PERIOD` is enabled by default if the Swift Pair advertising data provider is enabled in the configuration.

Force power down on bonded peer power off
-----------------------------------------

> The :kconfig:option:`CONFIG_CAF_BLE_ADV_POWER_DOWN_ON_DISCONNECTION_REASON_0X15` Kconfig option can be used to force power down when a bonded peer disconnects with reason ``0x15`` (Remote Device Terminated due to Power Off).

Reviewer: ```suggestion
You can use the :kconfig:option:`CONFIG_CAF_BLE_ADV_POWER_DOWN_ON_DISCONNECTION_REASON_0X15` Kconfig option to force power down when a bonded peer disconnects with reason ``0x15`` (Remote Device Terminated due to Power Off).
```

> The module instantly stops Bluetooth LE advertising and submits :c:struct:`force_power_down_event` on :c:struct:`ble_peer_event` with :c:member:`ble_peer_event.reason` of ``0x15`` and :c:member:`ble_peer_event.state` of :c:enum:`PEER_STATE_DISCONNECTED`.

Reviewer 2: ```suggestion
The module instantly stops Bluetooth LE advertising and performs the following:

 *  Force power down using :c:struct:`force_power_down_event` on Bluetooth LE peer event (:c:struct:`ble_peer_event`) with reason ``0x15`` (:c:member:`ble_peer_event.reason`).
 * Disconnects (:c:enum:`PEER_STATE_DISCONNECTED`)  the Bluetooth stack from the remote peer (:c:member:`ble_peer_event.state`).
 
```
Hope my understanding is correct. Wanted this sentence to be a bit more descriptive (what the system does after the Bluetooth LE advertising stops).

> The feature can be used to prevent waking up bonded peer until activity on peripheral's end is detected.

Reviewer: ```suggestion
You can use this feature to prevent waking up bonded peer until activity on the peripheral is detected.
```
